File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class UsageDataPoint(BaseModel):
    date: str
    queries: int
    tokens_prompt: int
    tokens_completion: int
    avg_latency_ms: float


class UsageResponse(BaseModel):
    data: list[UsageDataPoint]
    total_queries: int
    total_tokens: int


class TopQuery(BaseModel):
    query: str
    count: int


class TopQueriesResponse(BaseModel):
    queries: list[TopQuery]

File: frontend/src/types/analytics.ts
export interface UsageDataPoint {
  date: string;
  queries: number;
  tokens_prompt: number;
  tokens_completion: number;
  avg_latency_ms: number;
}

export interface UsageResponse {
  data: UsageDataPoint[];
  total_queries: number;
  total_tokens: number;
}

export interface TopQuery {
  query: string;
  count: number;
}

export interface TopQueriesResponse {
  queries: TopQuery[];
}
